### PR TITLE
fix(chat): fix stream_chat to return generator

### PR DESCRIPTION
**Pull request by zhjunqin**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

The "stream_chat" just return an tuple like (response, history), but stream_chat was supposed to return generator. 

You can refer to function ”def stream_chat“ in https://huggingface.co/THUDM/chatglm-6b/blob/main/modeling_chatglm.py#L1316  return "yield response, new_history"

And project https://github.com/imClumsyPanda/langchain-ChatGLM expect function ”def stream_chat“ to return a generator.

But the implementation in chatglm-6b reuse many logic in huggingface transformers function "def generate" in 
https://github.com/huggingface/transformers/blob/main/src/transformers/generation/utils.py#L1161.

Here juse use a queue to receive responses to be simple.

## Modification

Change  "stream_chat"  to return a generator.

## BC-breaking (Optional)

Change  "stream_chat"  to return a generator.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by kkscilife** [comment]
There is no check, please commit new content to add checks before merged.

**Comment by zhjunqin** [comment]
> There is no check, please commit new content to add checks before merged.

Could you help specify which kind of check? Thanks.
I tried to add "pre-commit" to project, but I find there are so many existing codes to change.

**Comment by kkscilife** [comment]
> > There is no check, please commit new content to add checks before merged.
> 
> Could you help specify which kind of check? Thanks. I tried to add "pre-commit" to project, but I find there are so many existing codes to change.

The check includes lint check and pr test. You can try sync fork you own branch fix-stream-chat from [InternLM:main](https://github.com/InternLM/InternLM/tree/main)

**Comment by zhjunqin** [comment]
> > > There is no check, please commit new content to add checks before merged.
> > 
> > 
> > Could you help specify which kind of check? Thanks. I tried to add "pre-commit" to project, but I find there are so many existing codes to change.
> 
> The check includes lint check and pr test. You can try sync fork you own branch fix-stream-chat from [InternLM:main](https://github.com/InternLM/InternLM/tree/main)

I sync my branch with [InternLM:main] and then update code by using "pre-commit run --all-files", please help check, thank.

**Comment by kkscilife** [comment]
> > > > There is no check, please commit new content to add checks before merged.
> > > 
> > > 
> > > Could you help specify which kind of check? Thanks. I tried to add "pre-commit" to project, but I find there are so many existing codes to change.
> > 
> > 
> > The check includes lint check and pr test. You can try sync fork you own branch fix-stream-chat from [InternLM:main](https://github.com/InternLM/InternLM/tree/main)
> 
> I sync my branch with [InternLM:main] and then update code by using "pre-commit run --all-files", please help check, thanks.

I suggest you can just sync without using "pre-commit run --all-files". If  using "pre-commit run --all-files", you pull request will change many files.

**Comment by kkscilife** [comment]
> > > > > > There is no check, please commit new content to add checks before merged.
> > > > > 
> > > > > 
> > > > > Could you help specify which kind of check? Thanks. I tried to add "pre-commit" to project, but I find there are so many existing codes to change.
> > > > 
> > > > 
> > > > The check includes lint check and pr test. You can try sync fork you own branch fix-stream-chat from [InternLM:main](https://github.com/InternLM/InternLM/tree/main)
> > > 
> > > 
> > > I sync my branch with [InternLM:main] and then update code by using "pre-commit run --all-files", please help check, thanks.
> > 
> > 
> > I suggest you can just sync without using "pre-commit run --all-files". If using "pre-commit run --all-files", you pull request will change many files.
> 
> Just sync with [InternLM:main](https://github.com/InternLM/InternLM/tree/main) and update again, please help review, thanks.

The basic check has been passed. Please wati for  [Harold-lkk](https://github.com/Harold-lkk) to review and merge

File: tools/transformers/modeling_internlm.py
Comment (Harold-lkk) [inline tools/transformers/modeling_internlm.py:852]: May rename to stream_producer

File: tools/transformers/modeling_internlm.py
Comment (Harold-lkk) [inline tools/transformers/modeling_internlm.py:866]: Put it bellow the `consumer()`

File: tools/transformers/modeling_internlm.py
Comment (Harold-lkk) [inline tools/transformers/modeling_internlm.py:873]:
```suggestion
            while True:
                res = response_queue.get()
                if res is not None:
                    yield res
                
```

File: tools/transformers/modeling_internlm.py
Comment (zhjunqin) [reply tools/transformers/modeling_internlm.py]: Fixed.

File: tools/transformers/modeling_internlm.py
Comment (zhjunqin) [reply tools/transformers/modeling_internlm.py]: Fixed.

File: tools/transformers/modeling_internlm.py
Comment (zhjunqin) [reply tools/transformers/modeling_internlm.py]: Fixed.

File: tools/transformers/modeling_internlm.py
Comment (zhjunqin) [reply tools/transformers/modeling_internlm.py]: Fixed.

File: tools/transformers/modeling_internlm.py
Comment (zhjunqin) [reply tools/transformers/modeling_internlm.py]: Fixed.

**Review by Harold-lkk — APPROVED** [review]

**Comment by zhjunqin** [comment]
@Harold-lkk 
There are 2 workflows still awaiting approval
This workflow requires approval from a maintainer. [Learn more about approving workflows.

Could you help check?

**Review by gaoyang07 — APPROVED** [review]

**Pull request merged**
